xylose/scielodocument.py

```diff
@@ -1913,11 +1965,27 @@ def review_date(self):
     @property
     def ahead_publication_date(self):
         """
-        This method retrieves the ahead of print date of the given article, if it exist.
-        This method deals with the legacy fields (223).
-        """
-        if 'v223' in self.data['article']:
-            return tools.get_date(self.data['article']['v223'][0]['_'])
+        This method retrieves the ahead of print date of the given article,
+        if it exist.
+        Since SPS 1.8.1 e 1.9 all the documents have online publication date
+        (v223).
+        This method returns document_publication_date, only if it is an
+        AOP article or an ex AOP
+        """
+        def normalize_page_number(page_number):
```

Comment:
@gustavofonseca não é isso.
Esta função é para retornar `None` se  as páginas que foram informadas ser igual a "0" ou "00", como era prática para dar página para _ahead of print_. Caso contrário retorna o mesmo valor.

Reply:
@gustavofonseca 
Estou pensando o seguinte:
Talvez o código possa ficar simplesmente:
```
        if self.issue.number and self.issue.number.lower() == 'ahead':
           return self.document_publication_date
```

no lugar de 
```
        if self.volume and \
                self.issue.number and self.issue.number.lower() == 'ahead':
            if not _bool(self.start_page) and \
               not _bool(self.end_page):
                return self.document_publication_date

```

Reply:
O nome desta função não reflete o que ela faz. O cliente desta função (linhas 1986 e 1987) está interessado num valor booleano, e ela entrega esse valor usando uma lógica que eu não chamaria de normalização. Basicamente retorna `False` para qualquer objeto de contexto booleano igual a `False`; retorna `True` para qualquer string diferente de `"0"`, exceto string vazia; levanta `AttributeError` nos demais casos. Acho que ficaria mais simples e robusto algo como:

```python
def _bool(value):
    """Implementação da função bool que também considera `False` 
    as strings de dígitos zero `"0"`.
    """
    try:
        return bool(int(value))
    except TypeError:
        return bool(value)
```